src/lib/sidenav/drawer.ts

```diff
@@ -359,6 +383,32 @@ export class MdDrawerContainer implements AfterContentInit {
         this._watchDrawerPosition(drawer);
       });
     });
+
+    const resize = typeof window !== 'undefined' ?
+      auditTime.call(fromEvent(window, 'resize'), 150) :
+      observableOf(null);
+
+    this._updateDrawer = startWith.call(resize, null).subscribe(() => {
+      if (this._element.nativeElement.offsetWidth < this.breakpointWidth) {
+        if (this.breakpointChangeMode) {
+          this._drawers.forEach(drawer => drawer.mode = 'over');
+        }
+        this.close();
+      } else {
+        if (this.breakpointChangeMode) {
+          this._drawers.forEach(drawer => drawer.mode = 'side');
+        }
+        this.open();
+      }
+    });
+
+  }
+
+  ngOnDestroy() {
+    if (this._updateDrawer) {
+      this._updateDrawer.unsubscribe();
+      this._updateDrawer = null;
```

Comment:
I don't think setting to `null` is necessary since the component is being destroyed